test/src/e2e_vm_tests/test_programs/should_fail/binop_intrinsics/test.toml

```diff
@@ -2,19 +2,53 @@ category = "fail"
 
 # check: $()error
 # check: $()__add(A { a: 32 }, 32);
-# nextln: $()Unsupported argument type to intrinsic "add"
+# nextln: $()Mismatched types
+# nextln: $()expected: A
+# nextln: $()found:    numeric.
+# nextln: $()help: Variable declaration's type annotation does not match up with the assigned expression's type.
+
+# check: $()error
+# check: $()__add(A { a: 32 }, 32);
+# nextln: $()Mismatched types
+# nextln: $()expected: numeric
+# nextln: $()found:    A.
+
+# check: $()error
+# check: $()__add("Hello", 22);
+# nextln: $()Mismatched types
+# nextln: $()expected: str[5]
+# nextln: $()found:    numeric.
```

Comment:
This looks confusing: `__add` should not expect a string type, it only supports numeric types according to the docs